velox/type/Conversions.h

```diff
@@ -37,6 +37,24 @@ struct Converter {
   }
 };
 
+namespace {
+inline void checkNoWhitespace(const char* data, size_t size) {
+  if (size > 0) {
+    char first = data[0];
+    char last = data[size - 1];
+    VELOX_USER_CHECK(
```

Comment:
@gggrace14 Throwing in CAST is going to cause a performance issue. See https://velox-lib.io/blog/optimize-try_cast./ for more details.

Let's find a better way. CC: @laithsakka

Reply:
Hi @mbasmanova , I got it. Thanks a lot for the pointer to the context of the perf issue! Going through it. Need to figure it out together with @laithsakka